lib/puppet/type/package.rb

```diff
@@ -20,6 +20,11 @@ module Puppet
       requires in order to function, and you must meet those requirements
       to use a given provider.
 
+      Multiple resources for packages from different providers that share
+      the same name can be specified, but they must have unique titles.
+      Resources referenced by Package[name] are no longer recognized as
+      they would be ambiguous, and must now be referenced by Package[title].
```

Comment:
We can tighten this up a bit by removing passive voice and focusing on the action and its consequences:

    You can declare multiple package resources with the same `name`, as long as they
    specify different providers and have unique titles.

    Note that you must use the _title_ to make a reference to a package resource;
    `Package[<NAME>]` is not a synonym for `Package[<TITLE>]` like it is for many
    other resource types.